dPhysics/dpEntity.cpp

```diff
@@ -76,16 +77,25 @@ void dpEntity::CheckCollision(dpEntity* other) {
 		return;
 	}
 
-	bool wasFound = m_CurrentlyCollidingObjects.contains(other->GetObjectID());
-
-	bool isColliding = m_CollisionShape->IsColliding(other->GetShape());
+	const auto objItr = std::find(
+		m_CurrentlyCollidingObjects.cbegin(),
+		m_CurrentlyCollidingObjects.cend(),
+		other->GetObjectID()
+	);
+	const bool wasFound = objItr != m_CurrentlyCollidingObjects.cend();
+	const bool isColliding = m_CollisionShape->IsColliding(other->GetShape());
 
 	if (isColliding && !wasFound) {
-		m_CurrentlyCollidingObjects.emplace(other->GetObjectID(), other);
-		m_NewObjects.push_back(other);
+		m_CurrentlyCollidingObjects.emplace_back(other->GetObjectID());
+		m_NewObjects.push_back(other->GetObjectID());
 	} else if (!isColliding && wasFound) {
-		m_CurrentlyCollidingObjects.erase(other->GetObjectID());
-		m_RemovedObjects.push_back(other);
+		// Erase object ID from currently colliding objects by swapping and popping vector
+		const auto objIdx = std::distance(m_CurrentlyCollidingObjects.cbegin(), objItr);
+		auto& lastObj = m_CurrentlyCollidingObjects.back();
+		m_CurrentlyCollidingObjects[objIdx] = std::move(lastObj);
+		m_CurrentlyCollidingObjects.pop_back();
```

Comment:
why if objItr not used?  it already is known to be the element we want to remove and has to exist to get into this if block, so it should be simpler to just do 
```cpp
*objItr = m_CurrentlyCollidingObjects.back();
m_CurrentlyCollidingObjects.pop_back();
```

Reply:
I am tired. This is a better solution. I will use it. 👍